src/components/TokenEnterAmount.test.tsx

```diff
@@ -79,141 +55,421 @@ describe('TokenEnterAmount', () => {
-      <Provider store={store}>
-        <TokenEnterAmount {...defaultProps} />
-      </Provider>
-    )
-    expect(getByTestId('tokenEnterAmount/TokenName')).toBeTruthy()
-    expect(getByText('CELO on Celo Alfajores')).toBeTruthy()
-    expect(getByTestId('tokenEnterAmount/SwitchTokens')).toBeTruthy()
-    expect(getByTestId('tokenEnterAmount/TokenSelect')).toBeTruthy()
-    expect(getByTestId('tokenEnterAmount/TokenBalance')).toBeTruthy()
-    expect(getByTestId('tokenEnterAmount/TokenBalance').props.children.props.i18nKey).toBe(
-      'tokenEnterAmount.availableBalance'
-    )
-  })
+    const replaceSeparators = (value: string) =>
+      value.replace(/\./g, '|').replace(/,/g, group).replace(/\|/g, decimal)
 
-  it('formats the input value correctly', () => {
-    const store = createMockStore(mockStore)
-    const { getByTestId, rerender } = render(
-      <Provider store={store}>
-        <TokenEnterAmount {...defaultProps} />
-      </Provider>
-    )
-    const input = getByTestId('tokenEnterAmount/TokenAmountInput')
-    const converted = getByTestId('tokenEnterAmount/ExchangeAmount')
-    expect(input.props.value).toBe('1')
-    expect(converted.props.children).toBe(`${APPROX_SYMBOL} $0.1`)
-    fireEvent.press(getByTestId('tokenEnterAmount/SwitchTokens'))
-
-    // simulate call of toggleAmountType
-    rerender(
-      <Provider store={store}>
-        <TokenEnterAmount
-          {...defaultProps}
-          amountType="local"
-          inputValue="0.1"
-          tokenAmount="1 CELO"
-        />
-      </Provider>
-    )
-
-    expect(input.props.value).toBe('₱0.1')
-    expect(converted.props.children).toBe(`${APPROX_SYMBOL} 1 CELO`)
-  })
+    it('properly groups numbers', () => {
+      expect(formatNumber('')).toBe('')
+      expect(formatNumber('123')).toBe(replaceSeparators('123'))
+      expect(formatNumber('1234')).toBe(replaceSeparators('1,234'))
+      expect(formatNumber('1234567')).toBe(replaceSeparators('1,234,567'))
+      expect(formatNumber('1234567.12345')).toBe(replaceSeparators('1,234,567.12345'))
+      expect(formatNumber('123456789012345')).toBe(replaceSeparators('123,456,789,012,345'))
+      expect(formatNumber('12.34567')).toBe(replaceSeparators('12.34567'))
+      expect(formatNumber('-1234567.89')).toBe(replaceSeparators('-1,234,567.89'))
+      expect(formatNumber('1234abc')).toBe(replaceSeparators('1,234abc'))
+      expect(formatNumber('1234.56xyz')).toBe(replaceSeparators('1,234.56xyz'))
+    })
 
-  it('handles input changes correctly', () => {
-    const store = createMockStore(mockStore)
-    const { getByTestId } = render(
-      <Provider store={store}>
-        <TokenEnterAmount {...defaultProps} inputValue="20" />
-      </Provider>
-    )
-
-    const input = getByTestId('tokenEnterAmount/TokenAmountInput')
-    fireEvent.changeText(input, '15')
-    expect(mockOnInputChange).toHaveBeenCalledWith('15')
-  })
+    it('properly rounds token amounts', () => {
```

Comment:
@kathaypacific Even knowing that these functions use `formatNumber` for groupings I think of it from the perspective of _"I need this function to return me these results regardless of the implementation"_. So in case we change the formatting function or we use some different logic for this function in the future - we can be sure sure that all of these use cases for rounding/formatting are precisely tested, regardless of what's under the hood.

Reply:
yes, that makes sense! the amount of test duplication made me wonder if it'd make sense to combine `getReadableLocalAmount`, `getReadableTokenAmount`, and `formatNumber` into something like `formatAmount` which returns the token and local amount. the functions seem to be used together now, i'm not sure if you have plans for them to be used separately later though

Reply:
@kathaypacific the only benefits I've considered when splitting it into separate functions are:
1. It makes it a lot easier to test each part independently and make sure that part is working as intended. This should make the debugging of the formatting a bit easier as we won't need to debug decimals/groupings/rounding/final formatting at the same time but rather just one of those.
2. To make it more readable. I could put them in a single function but IMHO it would make it more complex with a lot of `if`s and it would be harder to debug.

Reply:
@kathaypacific I guess in the end I've got it to a more [compromised form](https://github.com/valora-inc/wallet/pull/6243/files#diff-06ba8146966623b3c8c0a4b09c8f1c95a5aea055bc7d1450e06050aa01fa813cR55)!